test/components/modal.spec.tsx

```diff
@@ -0,0 +1,146 @@
+
+            await waitFor(() => expect(bodySelect('MODAL')).to.be.absent());
+        });
+    });
+
+    it('renders to the screen', async function() {
+        clientRenderer.render(<Modal isOpen={true} />);
+
+        await waitFor(() => expect(bodySelect('MODAL')!).to.be.present());
+    });
+
+    it('renders any children passed as props', async function() {
+        clientRenderer.render(
+            <Modal isOpen={true}>
+                <p data-automation-id="CHILD_1">child 1</p>
+                <p data-automation-id="CHILD_2">child 2</p>
+            </Modal>
+        );
+
+        await waitFor(() => {
+            expect(bodySelect('CHILD_1')).to.be.present();
+            expect(bodySelect('CHILD_2')).to.be.present();
+        });
+    });
+
+    it('takes the full width and height of the viewport and is centered in the viewport', async function() {
+        clientRenderer.render(<Modal isOpen={true} />);
+
+        function checkIfAlignedToScreen(element: Element) {
+            const rects = element.getBoundingClientRect();
+            return rects.top === 0 && rects.left === 0;
+        }
+
+        await waitFor(() => {
+            expect(checkIfAlignedToScreen(bodySelect('MODAL')!), 'The modal wasn\'t centered').to.equal(true);
+            expect(bodySelect('MODAL')!.clientHeight).to.equal(window.innerHeight);
+            expect(bodySelect('MODAL')!.clientWidth).to.equal(window.innerWidth);
+        });
+    });
+
+    it('renders one child in the center of the viewport', async function() {
+        clientRenderer.render(
+            <Modal isOpen={true}>
+                <p data-automation-id="CHILD_1">child 1</p>
+            </Modal>
+        );
+
+        await waitFor(() => {
+            const child = bodySelect('CHILD_1');
+            const modal = bodySelect('MODAL');
+            expect([child, modal]).to.be.horizontallyAligned('center', 1);
+            expect([child, modal]).to.be.verticallyAligned('center', 1);
+        });
+    });
+
+    it('renders children in horizontal alignment', async function() {
+        clientRenderer.render(
+            <Modal isOpen={true}>
+                <p data-automation-id="CHILD_1">child 1</p>
+                <p data-automation-id="CHILD_2">child 2</p>
+            </Modal>
+        );
+
+        await waitFor(() => {
+            const childOne = bodySelect('CHILD_1');
+            const childTwo = bodySelect('CHILD_2');
+            const modal = bodySelect('MODAL');
+            expect([childOne, childTwo, modal]).to.be.horizontallyAligned('center', 1);
+        });
+    });
+
+    it('adds overflow: hidden to the body when opened and removes it when closed', async function() {
+        clientRenderer.render(<Modal isOpen={true} />);
+
+        await waitFor(() => {
+            expect(window.getComputedStyle(document.body).overflow).to.equal('hidden');
+        });
+
+        clientRenderer.render(<Modal isOpen={false} />);
```

Comment:
This renders another "closed" modal and leaves the original one there. You need to get the container from clientRenderer and render this modal in the same container for it to update.

Reply:
Also relevant for other tests, thanks